tls/s2n_handshake_io.c

```diff
@@ -1060,26 +1073,36 @@ int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status * blocked)
 
         if (ACTIVE_STATE(conn).writer == this) {
             *blocked = S2N_BLOCKED_ON_WRITE;
-            if (s2n_handshake_write_io(conn) < 0 && s2n_errno != S2N_ERR_BLOCKED) {
-                /* Non-retryable write error. The peer might have sent an alert. Try and read it. */
-                const int write_errno = errno;
-                const int write_s2n_errno = s2n_errno;
-                const char *write_s2n_debug_str = s2n_debug_str;
-
-                if (s2n_handshake_read_io(conn) < 0 && s2n_errno == S2N_ERR_ALERT) {
-                    /* s2n_handshake_read_io has set s2n_errno */
-                    S2N_ERROR_PRESERVE_ERRNO();
-                } else {
-                    /* Let the write error take precedence if we didn't read an alert. */
-                    errno = write_errno;
-                    s2n_errno = write_s2n_errno;
-                    s2n_debug_str = write_s2n_debug_str;
-                    S2N_ERROR_PRESERVE_ERRNO();
+            const int r = s2n_handshake_write_io(conn);
```

Comment:
Can we use something like `write_result` instead of `r`?

Reply:
Sure